Simplify interrupt handling

TMRh20 opened the pull request:
- Simplify interrupt handling for RF24Gateway:
  a: Change interrupt to toggle a variable instead of actually handling radio data
  b: Move radio handling into the poll() function which is called anyway from the main loop
- Update examples to remove interrupt enable/disabling
- Formatting

TMRh20 committed:
Simplify interrupt handling

- Simplify interrupt handling for RF24Gateway
- Update examples to remove interrupt enable/disabling

TMRh20 committed:
Remove blank line

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

TMRh20 committed:
Remove blank line

TMRh20 committed:
Merge branch 'IRQ-FIxes' of https://github.com/nRF24/RF24Gateway into IRQ-FIxes

2bndy5 commented on RF24Gateway.cpp:
```diff
@@ -207,15 +206,15 @@ int ESBGateway<mesh_t, network_t, radio_t>::allocateTunDevice(char* dev, int fla
     // Create device
     if (ioctl(fd, TUNSETIFF, (void*)&ifr) < 0) {
-        //close(fd);
+        // close(fd);
```
Are all these comment changes from clang-format? I thought I configured it to allow for any/no spaces after single line comments.

> [!TIP]
> Use the .clang-format config `file` automatically (with optional glob file syntax):
> ```
> clang-format-14 -i -style=file RF24*.*
> ```
> The `-i` modifies files in-place instead of piping changes to stdout.

TMRh20 replied:
Yup, I just formatted using clang-format 14.0.6 and it did this. I assumed it just picks up the style file in RF24Gateway folder automatically.

2bndy5 replied:
It might've in v12 before, but there are significant (and rather un-noticable) changes between major versions of clang. I think without the `-style=file`, clang-format defaults to `llvm` style (which is super strict by comparison to our config).

2bndy5 replied:
Lucky that comment changes were practically all that it wanted to do.

BTW, github won't let us (at cpp-linter project) make a suggestion to remove lines from a diff. That's why those suggestions seems lacking.

2bndy5 commented on RF24Gateway.cpp:
```diff
@@ -667,7 +638,7 @@ void ESBGateway<mesh_t, network_t, radio_t>::handleTX()
     if (msg->size > MAX_PAYLOAD_SIZE)
     {
-        //printf("*****WTF OVER *****");
+        // printf("*****WTF OVER *****");
```
# 🤣 
"WTF OVER" sounds like something I would add after 30 minutes of debugging.

2bndy5 commented on examples/gwNodeInt/RF24GatewayNodeInt.cpp:
```diff
@@ -83,11 +82,9 @@ int main(int argc, char** argv)
             mesh.begin();
         }
 
-        gw.interrupts(); // Re-enable interrupts when done accessing the radio
```
This kind of API change implies a feature (aka minor) version bump. We should probably add something to CMakeLists.txt that checks RF24 lib version. CMake's [`find_library()`](https://cmake.org/cmake/help/latest/command/find_library.html) doesn't seem to care about the found lib's version, so I might have to dig deeper into how to mandate a certain RF24 version -- `find_library()`'s `VALIDATOR` arg (new in CMake v3.25) seems like a strong lead.

BTW, I was thinking this new Linux kernel IRQ support would be released under RF24 v1.5.0.

TMRh20 replied:
Well most users should be using the installer, using the latest versions, but I see where you're going and its probably a good idea. Although these changes should still work with previous RF24 versions in theory.

2bndy5 replied:
CMake's [`find_package()`](https://cmake.org/cmake/help/latest/command/find_package.html) has a [version verification](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-version-selection) if a required version is specified! 👀 I'm not sure if that will work with RF24 as a shared library though.

2bndy5 commented:
I see now why you were so enthusiastic about Linux kernel IRQ support. Hopefully they don't break our implementation of it anytime soon.

TMRh20 commented:
Haha, that was kind of an epic thread and coding session, but I think we are good for a while.

TMRh20 committed:
Modify delay

- Add a longer delay similar to the polling example to minimize CPU usage

TMRh20 committed:
Only delay if no action

- Only delay if no IRQ and no radio data available

TMRh20 committed:
Make delay configurable by users

- When using interrupts, still use the old waitDelay variable to set the delay in the poll() function

2bndy5 reviewed (approved):
I'll experiment with the version checking in CMake during a different PR.

2bndy5 commented:
cherry-picked this change to v1.x branch
